src/views/TradingCompetition/components/RegisterModal/RegisterWithProfile.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import styled from 'styled-components'
+import { Button, Heading, Text, Flex, Checkbox, AutoRenewIcon } from '@pancakeswap-libs/uikit'
+import { useTradingCompetitionContract } from 'hooks/useContract'
+import useI18n from 'hooks/useI18n'
+import { useToast } from 'state/hooks'
+import { CompetitionProps } from '../../types'
+
+const StyledCheckbox = styled(Checkbox)`
+  min-width: 24px;
+`
+
+const StyledLabel = styled.label`
+  cursor: pointer;
+`
+
+const RegisterWithProfile: React.FC<CompetitionProps> = ({ profile, onDismiss, onRegisterSuccess }) => {
+  const [isAcknowledged, setIsAcknowledged] = useState(false)
+  const [isConfirming, setIsConfirming] = useState(false)
+  const tradingCompetitionContract = useTradingCompetitionContract()
+  const { account } = useWeb3React()
+  const { toastSuccess, toastError } = useToast()
+  const TranslateString = useI18n()
+
+  const handleConfirmClick = () => {
+    tradingCompetitionContract.methods
+      .register()
+      .send({ from: account })
+      .on('sending', () => {
```

Comment:
Just a note. I found out that technically `on` doesn't have the `sending` event although it still seems to work.

`once` does however.